model/classes/ResultService.php

```diff
@@ -0,0 +1,131 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA
+ */
+
+namespace oat\taoLtiConsumer\model\classes;
+
+use oat\taoDelivery\model\container\delivery\AbstractContainer;
+use oat\taoDelivery\model\container\execution\ExecutionClientContainer;
+use oat\taoDelivery\model\container\ExecutionContainer;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use IMSGlobal\LTI\ToolProvider\ToolConsumer;
+use oat\oatbox\session\SessionService;
+use oat\generis\model\OntologyAwareTrait;
+use oat\taoLti\models\classes\LtiProvider\LtiProvider;
+use oat\taoLti\models\classes\LtiProvider\LtiProviderService;
+
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+/**
+ * Class LtiDeliveryContainer
+ *
+ * A delivery container to manage LTI based delivery
+ */
+class ResultService
+{
+    use ServiceLocatorAwareTrait;
+    //    use LoggerAwareTrait;
+
+    const SERVICE_ID = 'result_service';
+    const FAILURE_MESSAGE = 'failure';
+    const SUCCESS_MESSAGE = 'success';
+    const RESPONSE_TEMPLATE = '<?xml version="1.0" encoding="UTF-8"?>
+        <imsx_POXEnvelopeResponse xmlns="http://www.imsglobal.org/services/ltiv1p1/xsd/imsoms_v1p0">
+            <imsx_POXHeader>
+                <imsx_POXResponseHeaderInfo>
+                    <imsx_version>V1.0</imsx_version>
+                    <imsx_messageIdentifier>{{messageId}}</imsx_messageIdentifier>
+                    <imsx_statusInfo>
+                        <imsx_codeMajor>{{codeMajor}}</imsx_codeMajor>
+                        <imsx_severity>status</imsx_severity>
+                        <imsx_description>{{description}}</imsx_description>
+                        <imsx_messageRefIdentifier>{{messageIdentifier}}</imsx_messageRefIdentifier>
+                        <imsx_operationRefIdentifier>replaceResult</imsx_operationRefIdentifier>
+                    </imsx_statusInfo>
+                </imsx_POXResponseHeaderInfo>
+            </imsx_POXHeader>
+            <imsx_POXBody>
+                <replaceResultResponse />
+            </imsx_POXBody>
+        </imsx_POXEnvelopeResponse>
+    ';
+    const SCORE_DESCRIPTION_TEMPLATE = 'Score for {{sourceId}} is now {{score}}';
+
+    private $dom;
+    public static $statuses = array(
+        400 => 'Invalid score',
+        404 => 'DeliveryExecution not found',
+        501 => 'Method not implemented (for other requests than replaceResultRequest)',
+        500 => 'Other internal error => client should retry',
+    );
+
+    /**
+     * @param $payload string
+     * @return array|null
```

Comment:
This is disturbing:
- on success, it returns null
- on failure, it returns an array
I'd go by catching the exception on the controller, and then generate a return response with the given message in the ValidateResultRequest method.